streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java

```diff
@@ -112,24 +121,106 @@ public void shutdown() {
     }
 
     @Test
-    public void shouldRestoreState() throws ExecutionException, InterruptedException {
+    public void shouldRestoreStateFromSourceTopic() throws InterruptedException, IOException {
```

Comment:
nit: simplify `InterruptedException, IOException` to `Exception`

Reply:
I disagree. It can be nicer for the reader to know which checked exceptions we're potentially going to throw.

But it doesn't matter much, so do what you want ;)

Reply:
For API calls I agree, but for testing, I think to collapse to `Exception` is ok.

Reply:
For regular API calls (ie, "main code") we should list them. For a test, no exception should ever be thrown, and if, the test fails. Which exception is irrelevant IMHO.